dashboard/pages/fragment_graph.tsx

```diff
@@ -171,18 +182,31 @@ const SIDEBAR_WIDTH = 200
 type BackPressureAlgo = "p50" | "p90" | "p95" | "p99"
 const backPressureAlgos: BackPressureAlgo[] = ["p50", "p90", "p95", "p99"]
 
+type BackPressureDataSource = "Embedded" | "Prometheus"
+const backPressureDataSources: BackPressureDataSource[] = [
+  "Embedded",
+  "Prometheus",
+]
+
 export default function Streaming() {
   const { response: relationList } = useFetch(getStreamingJobs)
   const { response: fragmentList } = useFetch(getFragments)
 
   const [relationId, setRelationId] = useQueryState("id", parseAsInteger)
   const [backPressureAlgo, setBackPressureAlgo] = useQueryState("backPressure")
   const [selectedFragmentId, setSelectedFragmentId] = useState<number>()
+  // used to get the data source
+  const [backPressureDataSourceAlgo, setBackPressureDataSourceAlgo] =
+    useState("Embedded")
+  // used to set the back pressure choice list
+  const [backPressureAlgoOptions, setBackPressureAlgoOptions] = useState(
```

Comment:
Does it have to be a state? I think it can be derived from `backPressureDataSourceAlgo`.

Reply:
Good point, refactored.